src/main/java/org/wso2/carbon/inbound/salesforce/poll/BayeuxParameters.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2016, salesforce.com, inc. All rights reserved. Licensed under the BSD 3-Clause license. For full
+ * license text, see LICENSE.TXT file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+package org.wso2.carbon.inbound.salesforce.poll;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+/**
+ * @author hal.hildebrand
+ * @since 202
+ */
+public interface BayeuxParameters {
+
+    /**
+     * @return the bearer token used to authenticate
+     */
+    String bearerToken();
+
+    /**
+     * @return the URL of the platform Streaming API endpoint
+     */
+    default URL endpoint() {
+
+        String path = new StringBuilder().append(LoginHelper.COMETD_REPLAY).append(version()).toString();
+        try {
+            return new URL(host(), path);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException(
+                    String.format("Unable to create url: %s:%s", host().toExternalForm(), path), e);
+        }
+    }
+
+    default URL host() {
+
+        try {
+            return new URL(LoginHelper.LOGIN_ENDPOINT);
+        } catch (MalformedURLException e) {
+            throw new IllegalStateException(String.format("Unable to form URL for %s", LoginHelper.LOGIN_ENDPOINT));

```

Comment:
Throw e with your string otherwise we will loose the actual cause of the exception.